JASP-Engine/JASP/R/binomialtest.R

```diff
@@ -15,224 +15,315 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-BinomialTest <- function(dataset, options, perform, state, ...) {
+BinomialTest <- function(jaspResults, dataset, options, ...) {
 
-	variables <- unlist(options$variables)
+  # Set title
+  jaspResults$title <- "Binomial Test"
 
-	binomResults <- state$binomResults
-  if (is.null(binomResults))
-		binomResults <- .binomialTest(dataset, options, perform, variables)
+  # Read dataset
+  dataset <- .binomReadData(dataset, options)
 
-	descriptPlots <- state$descriptPlots
-  if (options[["descriptivesPlots"]] && is.null(descriptPlots))
-		descriptPlots <- .binomialDescriptivesPlot(dataset, options, perform, variables)
+  # Error checking
+  .binomCheckErrors(dataset, options)
 
-	results <- list()
-	results[["binomial"]] <- .BinomialTable(binomResults, options)
+  # Compute the results
+  binomResults <- .binomComputeResults(jaspResults, dataset, options)
 
-	if (! is.null(descriptPlots)) {
-		results[["descriptives"]] <- descriptPlots
-		results[["descriptives"]][["title"]] <- "Descriptive Plots"
-	}
+  # Output tables and plots
+  .binomTableMain(jaspResults, dataset, options, binomResults)
```

Comment:
you don't want to align the arguments of these three functions? :P

Reply:
I was unsure how to do it. If you have a suggestion I'll be happy to take it ;)

Reply:
Did this.